tests/batch_polymorphism_test.py

```diff
@@ -0,0 +1,58 @@
+import functools
+
+from absl.testing import absltest
+from absl.testing import parameterized
+import haiku as hk # type: ignore
```

Comment:
This is fine for an experiment (which it is), but we won't be able to submit this because we should not take a dependency on haiku.